### ISSUE TITLE
Pytorch: Utilize torch.compile

### ISSUE THREAD

**Issue by miili** [seisbench/seisbench#245]
pytorch 2.0 introduced `torch.compile`. It can speed up inference by 30%. Worth having a look

https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html

Compilation takes time though.

**Comment by yetinam** [comment]
I didn't try it myself yet, but from how I understand the interface, trying this out is probably as simple as running `model = torch.compile(model)` and then using `model.annotate(...)` as usual. As annotate calls the forward function internally, this should make use of the compiled model.

**Comment by miili** [comment]
Yes, this is what I figure. The compiled model can be further be tuned to optimize different aspects, coming at a cost of tradeoffs.

**Comment by yetinam** [comment]
That's great news! I think the best thing to do is add a "how to improve my model's speed" page to the documentation where we add these explanations. I'll have a go at this. The resampling issue is pretty unfortunate but probably not fixed that quickly. On a side note, are you doing your resampling in parallel? Obspy does not provide this functionality, but if you're anyhow doing this manually, that could give substantial improvements.

**Comment by yetinam** [comment]
In principle, vectorizing the resampling (and potential prefiltering) would be interesting, but I think it gets messy once traces aren't of the same length (what they usually aren't).
